File: String_Reverse/Reverse.java
/*
    Author : Pakshal Shashikant Jain 
    Date : 11/08/2023
    Program : Write a program which accept string from user and display it inn
              reverse order.

              Input : "MarvellouS"
              Output : "SuollevraM"
*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
import java.lang.*;
import java.util.*;

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class StrRev
{
    //Characteristics of Class
    String str;
    char arr[];

    //Constructor of Class
    StrRev() 
    {
        this.str = " ";
        this.arr = null ;
    }

    //Behaviour of Class To Take Input from user
    void Accept()
    {
        //Creating object of Scanner To Take Input from user
        Scanner sobj = new Scanner(System.in);

        //Taking Input from user
        System.out.println("Enter One String");
        this.str = sobj.nextLine();

        this.arr = new char[this.str.length()];

        this.arr = this.str.toCharArray();
    }

    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    //Behaviour of Class To Reverse String  
    void reverse()
    {
        int i = 0;  
        int start = 0;
        int end = 0;
        char temp = ' ';

        if(this.arr == null)
        {
            return;
        }

        //Logic To Reverse String 
        while(end != this.arr.length - 1)
        {
            end++;
        }

        while(start < end)
        {
            temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;

            start++;
            end--;
        }
    }
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Reverse
{
    //Main Function
    public static void main(String args[])
    {
        System.out.println("Jay Ganesh.......");

        //Creating object of StrRev Class 
        StrRev sobj = new StrRev();

        //Call To Behaviour of Class
        sobj.Accept();
        sobj.reverse(); 

        //Printing Reversed String
        System.out.print("Reversed String is :");
        System.out.print(sobj.arr);
    }
}

File: String_Reverse/Reverse.ts
/*
    Author : Pakshal Shashikant Jain 
    Date : 11/08/2023
    Program : Write a program which accept string from user and display it inn
              reverse order.

              Input : "MarvellouS"
              Output : "SuollevraM"
*/

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Reverse
{
    //Characteristics  of Class
    Str : string;
    arr : any[];

    //Constructor of Class
    constructor() 
    {
        this.Str = " ";
        this.arr = [];
    }

    //Behaviour of Class To Take Input from user
    Accept()
    {
        this.Str = "MarvellouS";
        this.arr = Array.from(this.Str);
    }

    //Behaviour of Class To Reverse String 
    Reverse()
    {
        var i : number = 0;
        var iStart : number = 0;
        var iEnd : number = 0;
        var temp : any = ' ';

        while(iEnd != this.arr.length)
        {   
            iEnd++;
        }

        while(iStart < iEnd)
        {
            temp = this.arr[iStart];
            this.arr[iStart] = this.arr[iEnd];
            this.arr[iEnd] = temp;

            iStart++;
            iEnd--;
        }
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
function main() 
{
    console.log("Jay Ganesh.....");

    //Creating object of Reverse Class 
    var robj = new Reverse();

    //Call To Behaviour of Class
    robj.Accept();
    robj.Reverse();

    //Printing Reversed String 
    process.stdout.write(robj.arr.join(""));
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Call To Main Function
main();